phantazm-api/src/main/java/com/github/phantazmnetwork/api/RayUtils.java

```diff
@@ -0,0 +1,120 @@
+package com.github.phantazmnetwork.api;
+
+import net.minestom.server.collision.Shape;
+import net.minestom.server.coordinate.Point;
+import net.minestom.server.coordinate.Pos;
+import net.minestom.server.coordinate.Vec;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class RayUtils {
+
+    private RayUtils() {
+        throw new UnsupportedOperationException();
+    }
+
+    @SuppressWarnings({"DuplicatedCode", "UnstableApiUsage"})
+    public static @NotNull Optional<Vec> rayTrace(@NotNull Shape shape, @NotNull Point boundingBoxPosition,
```

Comment:
also this should have a unit test

Reply:
hmm, good point, i don't think it would, and i don't really like the idea of allowing the origin to not be the minimum point, it introduces a bunch of other issues

Reply:
might have to keep it minestom-specific then